docs/verifier/query-builder.md

```diff
@@ -2,59 +2,209 @@
+
+### Circuit ID
+
+The Query Builder incorporates 6 [Circom](https://docs.circom.io/) circuits, each designed for specific verification scenarios. Selection of the appropriate Circuit ID depends on the verification context (off-chain or on-chain) and the proof type used when the claim was issued. Below is a guide to these circuits:
+
+1. **credentialAtomicQuerySigV2:** Verifies off-chain if a claim, signed by the Issuer, meets the verifier's criteria.
+2. **credentialAtomicQuerySigV2OnChain:** Similar to SIG V2, but for on-chain verification (smart contract verifiers).
+3. **credentialAtomicQueryMTPV2:** For off-chain verification of claims added to the issuer's claims Merkle Tree.
+4. **credentialAtomicQueryMTPV2OnChain:** Applies MTP V2 logic for on-chain scenarios (smart contract verifiers).
+5. **credentialAtomicQueryV3:** A versatile circuit for claims either signed by the Issuer or included in the Issuer's state, suitable for off-chain verification.
+6. **credentialAtomicQueryV3OnChain:** The on-chain counterpart of V3, also accommodating both SIG and MTP proof types.
+
+#### Circuit Selection Based on Proof Type
+
+The available circuits in the Query Builder are filtered based on the selected Proof Type:
+
+- **Signature-based (SIG) Proof Type:** For SIG, the list includes V2 SIG circuits (1 and 2) and the V3 circuits, as they support both SIG and MTP proof types.
+- **Merkle Tree Proof (MTP) Proof Type:** With MTP, the options presented are V2 MTP circuits (3 and 4) and also the V3 circuits, compatible with both SIG and MTP.
+
+It's important to consider the proof type as it directly influences the circuit options provided, ensuring alignment with the verification method (off-chain or on-chain via a smart contract) and the specific requirements of the query.
+
+:::caution
+V3 circuits are experimental as of this writing, so users should exercise caution and stay informed about their development and stability.
+:::
+
+### Proof of Uniqueness
+
+The Proof of Uniqueness enhances security and integrity in contexts requiring [sybil resistance](https://en.wikipedia.org/wiki/Sybil_attack). This feature is optional and it's exclusively supported by v3 circuits. It's managed through two fields in the Query Builder form:
+
+1. **Enable Proof of Uniqueness:** This checkbox is initially unchecked. When selected, it enables the Proof of Uniqueness feature.
+
+2. **Nullifier Session ID:** A numeric input field that remains disabled unless the "Check proof of uniqueness" checkbox is selected. It requires the user to provide a nullifier session ID.
+
+The nullifier session ID field is a critical component for calculating the SybilID, which is essential for establishing Proof of Uniqueness in a query. This ID serves as a verifier-specific session identifier, which might represent, for example, a voting ID. It is used in the SybilID calculation formula:
```

Comment:
let's rename SybilID to NullfierID (we don't use term sybil publicly I guess)

Reply:
Done